From 4da42b3b8dcd9e2abb24751b252c180b577b4144 Mon Sep 17 00:00:00 2001
Message-Id: <4da42b3b8dcd9e2abb24751b252c180b577b4144.1399312709.git.erack@redhat.com>
From: Eike Rathke <erack@redhat.com>
Date: Mon, 5 May 2014 18:18:57 +0100
Subject: [PATCH] resolved fdo#78294 default null-date for document import is
 1899-12-30
MIME-Version: 1.0
Content-Type: multipart/mixed; boundary="------------erAck-patch-parts"

This is a multi-part message in MIME format.
--------------erAck-patch-parts
Content-Type: text/plain; charset=UTF-8; format=fixed
Content-Transfer-Encoding: 8bit


Change-Id: I9dfe76d422ad732a081442b95a995d8d395a098e
(cherry picked from commit 571cefca474e6b77d68f9fa31f805dcf692927fd)
---
 sc/source/ui/docshell/docsh3.cxx | 6 ++++++
 1 file changed, 6 insertions(+)


--------------erAck-patch-parts
Content-Type: text/x-patch; name="0001-resolved-fdo-78294-default-null-date-for-document-im.patch"
Content-Transfer-Encoding: 8bit
Content-Disposition: attachment; filename="0001-resolved-fdo-78294-default-null-date-for-document-im.patch"

diff --git a/sc/source/ui/docshell/docsh3.cxx b/sc/source/ui/docshell/docsh3.cxx
index c07913f..64f3aa7 100644
--- a/sc/source/ui/docshell/docsh3.cxx
+++ b/sc/source/ui/docshell/docsh3.cxx
@@ -427,6 +427,12 @@ void ScDocShell::InitOptions(bool bForLoading)      // called from InitNew and L
         // so it must not be taken from the global options.
         // Calculation settings are handled separately in ScXMLBodyContext::EndElement.
         aDocOpt.SetStdPrecision( SvNumberFormatter::UNLIMITED_PRECISION );
+
+        // fdo#78294 The default null-date if
+        // <table:null-date table:date-value='...' />
+        // is absent is 1899-12-30 regardless what the configuration is set to.
+        // Import filters may override this value.
+        aDocOpt.SetDate( 30, 12, 1899);
     }
 
     aDocument.SetDocOptions( aDocOpt );

--------------erAck-patch-parts--


